File: ecommerce_app/lib/urun.dart
class Urun {
  final int urunId;
  final String urunAdi;
  final String urunMarka;
  final String urunGorsel;
  final double urunSatisFiyati;
  final double? indirimliFiyat; 
  final String kategoriAdi;     
  final String aciklama;

  Urun({
    required this.urunId,
    required this.urunAdi,
    required this.urunMarka,
    required this.urunGorsel,
    required this.urunSatisFiyati,
    this.indirimliFiyat,
    required this.kategoriAdi,
    required this.aciklama,
  });


  factory Urun.fromJson(Map<String, dynamic> json) {
    return Urun(
      
      urunId: json['urunId'] ?? 0,
      urunAdi: json['urunAdi'] ?? 'İsimsiz Ürün',
      urunMarka: json['urunMarka'] ?? '',
      urunGorsel: json['urunGorsel'] ?? '',
      urunSatisFiyati: (json['urunSatisFiyati'] ?? 0).toDouble(),
      indirimliFiyat: json['indirimliFiyat'] != null 
          ? (json['indirimliFiyat']).toDouble() 
          : null,
          
      kategoriAdi: json['kategoriAdi'] ?? '',
      aciklama: json['aciklama'] ?? '',
    );
  }

  get urunYazar => null;
}

File: ECommerceBackEnd/ECommerceBackEnd/Models/Urun.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization; 

namespace ECommerceBackEnd.Models
{
    public class Urun
    {
        [Key]
        public int UrunId { get; set; }

        [Column(TypeName = "VARCHAR")]
        [StringLength(30)]
        public string? UrunAdi { get; set; }

        [Column(TypeName = "VARCHAR")]
        [StringLength(30)]
        public string? UrunMarka { get; set; }

        [Column(TypeName = "VARCHAR")]
        [StringLength(250)]
        public string? UrunYazar { get; set; }

        public short UrunStok { get; set; }

        public decimal UrunSatisFiyati { get; set; }
        public string? UrunDil { get; set; }
        public int? UrunSayfa { get; set; }
        public bool UrunStokDurum { get; set; }

        [Column(TypeName = "VARCHAR")]
        [StringLength(250)]
        public string? UrunGorsel { get; set; }

        public int KategoriID { get; set; }

        [JsonIgnore]
        public virtual Kategori? Kategori { get; set; }

        public bool Durum { get; set; }

        public decimal? IndirimliFiyat { get; set; }

        [Column(TypeName = "NVARCHAR(MAX)")]
        public string? Aciklama { get; set; }

        [JsonIgnore]
        public ICollection<Satislar>? satislars { get; set; }

        [NotMapped]
        public string? KategoriAdi { get; set; }
    }
}